### DISCUSSION TITLE
Want to add Grafana as a exporter in Opentelemetry collector

### DISCUSSION THREAD

**Discussion by sshiv5768** [open-telemetry/community#1386] [Community Discussions]
## Goal
Receive telemetry data from the simple flask app to opentelemetry collector and send it to Grafana.

## Approach
- Created a demo flask app.
- Getting telemetry data to Otel-Collector container
-  Running Grafana container
- Added Grafana credentials to the ``config.yml``

## Issues facing
- There is no direct plugin for opentelemetry in grafana.
- Getting below error when i run ``opentelemetry-instrument flask run``
```
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'exporters': unknown type: "grafana" for id: "grafana" (valid values: [logging jaeger kafka prometheusremotewrite zipkin otlp otlphttp file opencensus prometheus])
2023/03/17 06:46:07 collector server run finished with error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'exporters': unknown type: "grafana" for id: "grafana" (valid values: [logging jaeger kafka prometheusremotewrite zipkin otlp otlphttp file opencensus prometheus])
```

**Comment by Depechie** [comment]
@sshiv5768 Grafana is only the visual layer. If you want to view the data from your app, you need to configure a data source inside Grafana.

For logs that would be Loki.
For traces that would be Tempo.
And metrics will fit within Prometheus.

For each element you can configure an exporter in your collector config and that way you can setup dashboards in Grafana for the data in each data source.

**Comment by pyohannes** [comment]
@sshiv5768 For experimentation purposes, you could give this a try: https://github.com/grafana/docker-otel-lgtm

It provides a docker image that includes Loki, Tempo, and Prometheus. You can send data via the standard OTLP protocol.

**Reply by Depechie** [reply]
Similar setup but with .NET instead of Java can be found here [https://github.com/Depechie/OpenTelemetryGrafana](https://github.com/Depechie/OpenTelemetryGrafana) ;)
